### PR TITLE
identity/gitlab: fix retrieve group error

### LINKED ISSUE #612: Error 500 for self hosted gitlab. could not retrieve groups
## What happened?
After authentication, user is redirected to the page attached in the picture.

Screenshot from 2020-04-15 09-29-49

## What did you expect to happen?

Successful login and be able to view httpbin.

## How'd it happen?

1. Ran pomerium docker configuration.
2. I visited https://httpbin.localhost.pomerium.io, accepted my self signed cert
3. Was redirected to gitlab for auth. Successfully authenticated. 
4. Saw error 500: could not retrieve groups Unauthorized

## What's your environment like?

- Pomerium version (retrieve with `pomerium --version` or `/ping` endpoint): 0.7
- Server Operating System/Architecture/Cloud: Docker on local machine

## What's your config.yaml?

**Pomerium config**
```config.yaml
pomerium_debug: true
authenticate_service_url: https://authenticate.localhost.pomerium.io

certificate_file: "./cert.pem"
certificate_key_file: "./privkey.pem"

idp_provider: gitlab
idp_provider_url: "https://<snip>"
idp_scopes: "openid,read_user,profile,email"
idp_client_id: <snip>
idp_client_secret: <snip>

policy:
  - from: https://httpbin.localhost.pomerium.io
    to: https://httpbin.org
```

**Docker compose**
```config.yaml
version: "3"
services:
  pomerium:
    image: pomerium/pomerium:v0.7.0
    environment:
      - COOKIE_SECRET=<snip>
    volumes:
      - ./authenticate.localhost.pomerium.io.crt:/pomerium/cert.pem:ro
      - ./authenticate.localhost.pomerium.io.key:/pomerium/privkey.pem:ro
      - ./config.yaml:/pomerium/config.yaml:ro
    ports:
      - 443:443
```

## What did you see in the logs?

```logs
httputil: ErrorResponse error="Internal Server Error: oauth callback : error redeeming authenticate code: internal/identity: could not retrieve groups Unauthorized" ip=172.17.0.1
```

## Additional context
I had to add the `idp_scopes`and restrict it to `openid,read_user,profile,email` since the default ones also included the `api` scope. Why should the app need that? I tried giving it that scope anyway, allowing it in gitlab, but that made no difference, I still had the same error.

### PR DESCRIPTION
**Pull request by Lumexralph**
## Summary

 - remove hardcoded gitlab provider url.
 - update the gitlab identity provider documentation.
## Related issues

Fixes #612 

**Checklist**:
- [X] add related issues
- [X] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [X] ready for review

### CODE REVIEW

File: docs/docs/identity-providers/gitlab.md [hunk 1]
```diff
@@ -25,8 +25,19 @@ Field        | Description
 ------------ | --------------------------------------------
 Name         | The name of your web app
 Redirect URI | `https://${authenticate_service_url}/oauth2/callback`
-Scopes       | **Must** select **read_user** and **openid**
+Scopes       | **Must** select **openid**, **read_user** and **api**
 
+If no scopes are set, we will use the following scopes:
```
Comment (desimone) [inline docs/docs/identity-providers/gitlab.md:30]: The formatting of this list is a bit off. See : https://deploy-preview-614--pomerium-docs.netlify.app/docs/identity-providers/gitlab.html

What do you think about listing them in line like....
If no scopes are set, we will use the following scopes: `openid`, `api`, `read_user`,`profile`, and `email`.

File: internal/identity/gitlab.go [hunk 2]
```diff
@@ -19,8 +19,7 @@ import (
 
 const (
 	defaultGitLabProviderURL = "https://gitlab.com"
-	revokeURL                = "https://gitlab.com/oauth/revoke"
```
Comment (desimone) [inline internal/identity/gitlab.go:22]: this is now being built from the open-id connect well known rest endpoint now?

Comment (Lumexralph) [reply internal/identity/gitlab.go:22]: Yeah

File: internal/identity/gitlab.go [hunk 3]
```diff
@@ -90,7 +88,7 @@ func (p *GitLabProvider) UserGroups(ctx context.Context, s *sessions.State) ([]s
 		FullPath                       string      `json:"full_path,omitempty"`
 	}
 	headers := map[string]string{"Authorization": fmt.Sprintf("Bearer %s", s.AccessToken.AccessToken)}
-	err := httputil.Client(ctx, http.MethodGet, defaultGitLabGroupURL, version.UserAgent(), headers, nil, &response)
+	err := httputil.Client(ctx, http.MethodGet, p.ProviderURL+groupPath, version.UserAgent(), headers, nil, &response)
```
[the remark this answers is not in the document]
Comment (Lumexralph) [reply internal/identity/gitlab.go:91]: that makes sense to me, I will implement the feedback. Thanks.

**Pull request merged**